modules/Settings/LayoutEditor/actions/Field.php

```diff
@@ -324,4 +329,24 @@ public function createSystemField(App\Request $request): void
 		}
 		$response->emit();
 	}
+
+	/**
+	 * Check if column length has changed.
+	 *
+	 * @param Settings_LayoutEditor_Field_Model $fieldInstance
+	 * @param string                            $columnLength
+	 *
+	 * @return void
+	 */
+	private function changeColumnLength(Settings_LayoutEditor_Field_Model $fieldInstance, string $columnLength)
+	{
+		$uitypeModel = $fieldInstance->getUITypeModel();
+		if ($uitypeModel->isResizableColumn() && $uitypeModel->isColumnLengthIncreased($columnLength) && $uitypeModel->validateColumnLength($columnLength)) {
+			$dbColumnStructure = $fieldInstance->getDBColumnType(false);
+			$columnType = $dbColumnStructure['type'];
+			$db = App\Db::getInstance();
+			$db->createCommand()->alterColumn($fieldInstance->getTableName(), $fieldInstance->getColumnName(), "{$columnType}({$columnLength})")->execute();
+			$fieldInstance->set('maximumlength', $fieldInstance->getRangeValues());
+		}
```

Comment:
You won't always need to change the column type. By default, only maximumlength is to be modified, and the column type in special cases.